components/ai_chat/resources/page/components/conversation_list/index.tsx

```diff
@@ -15,11 +15,44 @@ import ContextMenuAssistant from '../context_menu_assistant'
 import { getLocale } from '$web-common/locale'
 import SiteTitle from '../site_title'
 
+const CodeBlock = React.lazy(async () => ({ default: (await import('../code_block')).default.Block }))
+const CodeInline = React.lazy(async () => ({ default: (await import('../code_block')).default.Inline }))
+
 const SUGGESTION_STATUS_SHOW_BUTTON: mojom.SuggestionGenerationStatus[] = [
   mojom.SuggestionGenerationStatus.CanGenerate,
   mojom.SuggestionGenerationStatus.IsGenerating
 ]
 
+const regexp = /```([^\n`]+)?\n?([\s\S]*?)```|`(.*?)`|([^`]+)/gs
+
+interface FormattedTextProps {
+  text: string
+}
+
+function FormattedTextRenderer(props: FormattedTextProps): JSX.Element {
```

Comment:
nit: Well, now that this is so simple, you could more simply just wrap this function in `React.memo`